library/alloc/src/string.rs

```diff
@@ -2012,6 +2062,30 @@ impl FromUtf8Error {
         &self.bytes[..]
     }
 
+    /// Converts the bytes into a `String` lossily, substituting invalid UTF-8
+    /// sequences with replacement characters.
+    ///
+    /// See [`String::from_utf8_lossy`] for more details on replacement of
+    /// invalid sequences, and [`String::from_utf8_lossy_owned`] for the
+    /// `String` function which corresponds to this function.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(string_from_utf8_lossy_owned)]
+    /// // some invalid bytes
+    /// let input: Vec<u8> = b"Hello \xF0\x90\x80World".into();
+    /// let output = String::from_utf8(input).unwrap_or_else(|e| e.into_utf8_lossy());
```

Comment:
this example isn't great, since it's just `String::from_utf8_lossy_owned(input)` but with more characters. I can't think of any better one though, so it's fine.

Reply:
I agree.

After some thinking, something more illustrative could be matching on the result of `String::from_utf8`

```rust
// some invalid bytes
let input: Vec<u8> = b"Hello \xF0\x90\x80World".into();

let output = match String::from_utf8(input) {
    Ok(s) => s,
    Err(e) => {
        // Log invalid UTF-8 error
        // ...
        
        // Lossily convert the bytes to a `String`
        e.into_utf8_lossy()
    }
};

assert_eq!(String::from("Hello �World"), output);
```

Reply:
that sounds better